ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -855,10 +868,32 @@ private static ConfigDef configDef(final ConfigGeneration generation) {
     return generation == ConfigGeneration.CURRENT ? CURRENT_DEF : LEGACY_DEF;
   }
 
+  private static DeploymentType parseDeploymentType(Object value) {
+    try {
+      return DeploymentType.valueOf(value.toString());
+    } catch (IllegalArgumentException e) {
+      throw new ConfigException(
+          "'" + KSQL_DEPLOYMENT_TYPE_CONFIG + "' must be one of: "
+          + Arrays.asList(DeploymentType.values())
+          + " however we found '" + value + "'"
+      );
+    }
+  }
+
   // CHECKSTYLE_RULES.OFF: MethodLength
   private static ConfigDef buildConfigDef(final ConfigGeneration generation) {
     final ConfigDef configDef = new ConfigDef()
         .define(
+            KSQL_DEPLOYMENT_TYPE_CONFIG,
+            ConfigDef.Type.STRING,
+            DeploymentType.confluent.name(),
```

Comment:
Shoudln't `selfManaged` be the default to keep the same behavior for self managed users that upgrade?